model/job/broker.go

```diff
@@ -313,61 +314,86 @@ func GetQueuedJobs(db prefixer.Prefixer, workerType string) ([]*Job, error) {
 	return results, nil
 }
 
-// GetJobsBeforeDate returns alls jobs queued before the specified date
-func GetJobsBeforeDate(db prefixer.Prefixer, date time.Time) ([]*Job, error) {
-	var cursor int
-	var jobs []*Job
+func GetAllJobs(db prefixer.Prefixer) ([]*Job, error) {
+	var startkey string
+	var lastJob *Job
 
 	finalJobs := []*Job{}
+	remainingJobs := true
 
-	for cursor != -1 {
-		jobs = []*Job{}
-		req := &couchdb.FindRequest{
-			UseIndex: "by-queued-at",
-			Selector: mango.Lt("queued_at", date.Format(time.RFC3339Nano)),
-			Limit:    1000,
-			Skip:     cursor,
+	for remainingJobs {
+		jobs := []*Job{}
+		req := &couchdb.AllDocsRequest{
+			Limit:    10001,
+			StartKey: startkey,
 		}
-		err := couchdb.FindDocs(db, consts.Jobs, req, &jobs)
+		err := couchdb.GetAllDocs(db, consts.Jobs, req, &jobs)
+
 		if err != nil {
 			return nil, err
 		}
+
+		lastJob, jobs = jobs[len(jobs)-1], jobs[:len(jobs)-1]
+
+		// Startkey for the next request
+		startkey = lastJob.JobID
+
+		// Appending to the final jobs
 		finalJobs = append(finalJobs, jobs...)
 
+		// Only the startkey is present: we are in the last lap of the loop
+		// We have to append the startkey as the last element
 		if len(jobs) == 0 {
-			cursor = -1
-		} else {
-			cursor = len(finalJobs)
+			remainingJobs = false
+			finalJobs = append(finalJobs, lastJob)
 		}
 	}
 
 	return finalJobs, nil
 }
 
+// GetJobsBeforeDate returns alls jobs queued before the specified date
+func GetJobsBeforeDate(jobs []*Job, date time.Time) ([]*Job, error) {
```

Comment:
Why do you return an `error` if it is always nil?

Reply:
This is a mistake, thanks !